### PR TITLE
Allow string literals to be used as identifiers

### PR DESCRIPTION
**Pull request by mehcode**
Allows for patterns like:

```js
const messages = defineMessages({
  'title': '...',
  'message': '...',
  'person[key]name': '...'
})
```

---

💯 Awesome babel plugin, by the way. Just need this to convert.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -30,10 +30,19 @@ const getPrefix = ({
 }
 
 const getId = (path, prefix) => {
-  if (!(path.isIdentifier() && path.node.name)) {
-    throw new Error(`require Object key`)
+  let name;
+  
+  if (path.isStringLiteral()) {
+    name = path.node.value
+  } else if (path.isLiteral()) {
```
Comment (akameco) [inline src/index.js]: `path.isLiteral()` → `path.isIdentifier()`

```js
const messages = defineMessages({
  title: '...', // ← Identifier
  'title2': '...' // ← StringLiteral
})
```


Please check this.
https://astexplorer.net/#/gist/28061859b30ad2cd7aef75ee70aed767/latest

**Review by akameco — CHANGES_REQUESTED** [review]
Thank you @mehcode :)

Could you fix this?

File: src/index.js (hunk 1 above)
Comment (mehcode) [reply src/index.js]: Sorry about that. Bad copy paste.

**Comment by akameco** [comment]
Thank you @mehcode 😀

**Pull request merged**